### PR TITLE
Add terminationGracePeriodSeconds support For container app

### PR DESCRIPTION
**Pull request by njuCZ**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/containerapp/azext_containerapp/custom.py [hunk 1]
```diff
@@ -881,6 +886,10 @@ def update_containerapp_logic(cmd,
         new_containerapp["properties"]["template"] = {} if "template" not in new_containerapp["properties"] else new_containerapp["properties"]["template"]
         new_containerapp["properties"]["template"]["revisionSuffix"] = revision_suffix
 
+    if termination_grace_period is not None:
+        new_containerapp["properties"]["template"] = {} if "template" not in new_containerapp["properties"] else new_containerapp["properties"]["template"]
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/custom.py:890]: Recommend to use `safe_set("properties", "template", "terminationGracePeriodSeconds", value=termination_grace_period)`

File: src/containerapp/azext_containerapp/_params.py [hunk 2]
```diff
@@ -120,6 +120,7 @@ def load_arguments(self, _):
         c.argument('traffic_weights', nargs='*', options_list=['--traffic-weight'], help="A list of revision weight(s) for the container app. Space-separated values in 'revision_name=weight' format. For latest revision, use 'latest=weight'")
         c.argument('workload_profile_name', options_list=['--workload-profile-name', '-w'], help="Name of the workload profile to run the app on.", is_preview=True)
         c.argument('secret_volume_mount', help="Path to mount all secrets e.g. mnt/secrets", is_preview=True)
+        c.argument('termination_grace_period', type=int, options_list=['--termination-grace-period', '--grace-period'], help="Duration in seconds a replica is given to gracefully shut down before it is forcefully terminated. (Default: 30)", is_preview=True)
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/_params.py:123]: is it support in --yaml?

Comment (Greedygre) [reply src/containerapp/azext_containerapp/_params.py:123]: The termination_grace_period already existed in  _sdk_models, so it is supported in --yaml.
@njuCZ  Could you please add some test for property termination_grace_period in test_containerapp_create_with_yaml or add new test case?

Comment (njuCZ) [reply src/containerapp/azext_containerapp/_params.py]: only api-version >= 2023-04-01-preview support this property. we are still using python sdk 2022-11-01-preview, so it's not supported in --yaml

Comment (Greedygre) [reply src/containerapp/azext_containerapp/_params.py]: you can manual add this property

Comment (njuCZ) [reply src/containerapp/azext_containerapp/_params.py]: updated

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 
Could you please help to review this pr?

File: src/containerapp/HISTORY.rst [hunk 3]
```diff
@@ -17,6 +17,7 @@ Upcoming
 * Add 'az containerapp ingress cors' for CORS support
 * 'az container app env create/update': support --enable-mtls parameter
 * 'az containerapp up': fix issue where --repo throws KeyError
+* 'az containerapp create/update': --termination-grace-period support custom termination grace period
```
Comment (zhoxing-ms) [inline src/containerapp/HISTORY.rst:20]: Please do not add the history note into a extension version which had already been released.

Comment (njuCZ) [reply src/containerapp/HISTORY.rst:20]: this means support a new property when az containerapp create/update, maybe I can change the description better

Comment (zhoxing-ms) [reply src/containerapp/HISTORY.rst:20]: If you want to release a new extension version for this PR, you need to upgrade the version defined in `HISTORY.rst` and `setup.py`. 
If you don't want to release a new extension version, you can put the history note under `Upcoming`

Comment (njuCZ) [reply src/containerapp/HISTORY.rst:20]: I rebase main branch to fix a conflict just now

Comment (njuCZ) [reply src/containerapp/HISTORY.rst]: it seems us team released a new version last night... Have updated it

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**